datasette/app.py

```diff
@@ -443,6 +443,71 @@ def __init__(
         self._root_token = secrets.token_hex(32)
         self.client = DatasetteClient(self)
 
+    async def initialize_internal_database(self):
+        await self.get_internal_database().execute_write_script(
+            """
+              CREATE TABLE IF NOT EXISTS datasette_metadata_instance_entries(
+                key text,
+                value text,
+                unique(key)
+              );
+
+              CREATE TABLE IF NOT EXISTS datasette_metadata_database_entries(
+                database_name text,
+                key text,
+                value text,
+                unique(database_name, key)
+              );
```

Comment:
This logic should live in `datasette/utils/internal_db.py` since that's where the other logic that creates tables and populates them lives.